test/metabase/driver/common/parameters/dates_test.clj

```diff
@@ -127,61 +130,70 @@
 (deftest date-string->range-test
   (t/with-clock (t/mock-clock #t "2016-06-07T12:13:55Z")
     (doseq [[group s->expected]
-            {"absolute datetimes"         {"Q1-2016"               {:end "2016-03-31", :start "2016-01-01"}
-                                           "2016-02"               {:end "2016-02-29", :start "2016-02-01"}
-                                           "2016-04-18"            {:end "2016-04-18", :start "2016-04-18"}
-                                           "2016-04-18~2016-04-23" {:end "2016-04-23", :start "2016-04-18"}
+            {"absolute datetimes"         {"Q1-2016"               {:start "2016-01-01", :end "2016-03-31"}
+                                           "2016-02"               {:start "2016-02-01", :end "2016-02-29"}
+                                           "2016-04-18"            {:start "2016-04-18", :end "2016-04-18"}
+                                           "2016-04-18~2016-04-23" {:start "2016-04-18", :end "2016-04-23"}
```

Comment:
👍 on changing the orders around here... doesn't make any difference if you're a computer but it's certainly weird to have to flip these around in your head to get a real date range when looking at it as a human. Thanks for doing this